### PR TITLE
ncurses: adding external support.

### PR DESCRIPTION
**Pull request by rblake-llnl**

### CODE REVIEW

File: var/spack/repos/builtin/packages/ncurses/package.py [hunk 1]
```diff
@@ -20,6 +22,8 @@ class Ncurses(AutotoolsPackage, GNUMirrorPackage):
     # URL must remain http:// so Spack can bootstrap curl
     gnu_mirror_path = "ncurses/ncurses-6.1.tar.gz"
 
+    executables = [r'ncurses\d*-config']
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ncurses/package.py]:
```suggestion
    executables = [r'^ncursesw?\d*-config$']
```
My Spack installation has both `ncurses6-config` and `ncursesw6-config`

File: var/spack/repos/builtin/packages/ncurses/package.py [hunk 2]
```diff
@@ -36,6 +40,39 @@ class Ncurses(AutotoolsPackage, GNUMirrorPackage):
     patch('patch_gcc_5.txt', when='@6.0%gcc@5.0:')
     patch('sed_pgi.patch',   when='@:6.0')
 
+    @classmethod
+    def determine_version(cls, exe):
+        output = Executable(exe)('--version', output=str, error=str)
+        match = re.search(r'^(\d+\S+)', output)
+        return match.group(1) if match else None
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ncurses/package.py]:
```suggestion
        return Executable(exe)('--version', output=str, error=str).rstrip()
```

File: var/spack/repos/builtin/packages/ncurses/package.py [hunk 3]
```diff
@@ -36,6 +40,37 @@ class Ncurses(AutotoolsPackage, GNUMirrorPackage):
...
+        return Executable(exe)('--version', output=str, error=str).rstrip()
+
+    @classmethod
+    def determine_variants(cls, exes, version):
+        results = []
+        for exe in exes:
+            variants = ''
+            output = Executable(exe)('--libs', output=str, error=str)
+
+            if re.search(r'-ltinfo', output):
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ncurses/package.py]:
```suggestion
            if '-ltinfo' in output:
```
This removes the `re` dependency

File: var/spack/repos/builtin/packages/ncurses/package.py [hunk 4]
```diff
@@ -7,6 +7,8 @@
 from glob import glob
 from os.path import exists, join
 from os import makedirs
+import re
+import os
```
Comment (adamjstewart) [inline var/spack/repos/builtin/packages/ncurses/package.py:8]: I think you can remove `re` now. You can also consolidate `os` if you want.

Comment (rblake-llnl) [reply var/spack/repos/builtin/packages/ncurses/package.py:8]: Standardized the inputs for clarity.

**Review by adamjstewart — APPROVED** [review]

**Pull request merged**